iOS/MVC/README.md

```diff
@@ -0,0 +1,46 @@
+
+2- you should not use one storyboard for your whole project . when you are using one storyboard for your whole project you will have these problems below :
+*you will have conflicts with your teammates when you want to work on the storyboard because you have one story board and even if you work in another part of that this one still is one file.
+*your story board will be very heavy to load and sometimes it's very hard to see the relation between the view controllers in the storyboard.
+
+### Network
+Getting the data from the server needs to be implemented in the models or separate class for calling the api's. network should not be called in the view controller directly. 
+
+### class sizes
+Your classes codes should not be more than 500 lines of codes it can be less or more, but if you follow the rules which mentioned above your classes codes will be much less than this (although it's not 0 or 1 and it depends on your logic but as soon as your classes sizes are small you are good to go)
+
+### Extensions and xib files 
+you can use extensions and xib files to avoid repeating yourself.
+Extensions add new functionality to an existing class, structure, enumeration, or protocol type. This includes the ability to extend types for which you don’t have access to the original source code (known as retroactive modeling). making each extension in separate file to implement and find the functions for the each extension easily 
+you can divide your complicated views to the smaller reusable views by using xib files, then you won't need to set a lot of constraints in the complicated views and each xib file is responsive so you can use them easily and anywhere. also for the `UICollectionViewCell` and `UITableViewCell` you can create classes with xib files instead of making them in the story boards.
+
+### Setting Constraints 
+the easiest and fastest way to setting the constraints is using the AutoLayout part in the storyboard and xib files without need to set any constraints in the codes (some times you need to change the constraints run time and the size will set in the runtime as well so in that case you need to set the constraints in the storyboard or xib files and then drag that constraint to your code to change it run time)
+
+### passing the data 
+to pass the data do not use `Notification Center` or `User Defaults`, you can use call back functions and delegate protocols instead . if you are using delegate protocols remember to make your delegates `weak`to avoid memory leaks. in some callback functions you need to use `weak` for avoiding the memory leaks problem. make sure every view controller which dismissed the `deinit` function executes in that class after dismissing. you can make some protocol functions optional to avoid using those functions in the other classes that you may use.  
```

Comment:
This one should break into small topics.

1. **Do not use `Notification Center` or `User Defaults`**. You can use call back functions and delegate protocols instead
2.  **If you are using delegate protocols remember to make your delegates `weak` to avoid memory leaks.** in some callback functions you need to use `weak` for avoiding the memory leaks problem.
3. **Make sure every view controller which dismissed the `deinit` function executes in that class after dismissing.** you can make some protocol functions optional to avoid using those functions in the other classes that you may use.

Reply:
this part explains how to pass the data , it doesn't contains separate parts .

Reply:
I agree with you. I just think that topic "pass the data" can be break down into smaller topics but it is ok to combine together.  

Here is sample on apple documentation on how they break down topics:
![image](https://user-images.githubusercontent.com/5284620/121118212-aade6e00-c843-11eb-8c49-88d6559b8c68.png)

Reply:
ok